### PR TITLE
Add documentation for caching issue

### PR DESCRIPTION
**Pull request by gabrielgiroe1**
Problem: cached rows on table do not update when record's associations suffer changes

### CODE REVIEW

File: docs/3.0/cache.md [hunk 1]
```diff
@@ -39,6 +39,14 @@ config.cache_store = ActiveSupport::Cache.lookup_store(:solid_cache_store)
 Our computed system do not use MemoryStore in production because it will not be shared between multiple processes (when using Puma).
 :::
 
+## Problem: cached rows on table do not update when record's associations suffer changes
+:::warning
+
```
Comment (Paul-Bob) [inline docs/3.0/cache.md:44]:
```suggestion
:::warning
```

File: docs/3.0/cache.md [hunk 2]
```diff
@@ -39,6 +39,14 @@ config.cache_store = ActiveSupport::Cache.lookup_store(:solid_cache_store)
...
+Solution 1: touch: true on association
+
+Solution 2: override cache_hash method on resource to take associations in consideration
+:::
+
```
Comment (Paul-Bob) [inline docs/3.0/cache.md:49]:
```suggestion
## Problem: cached rows on table do not update when record's associations suffer changes
:::warning

Solution 1: touch: true on association
Solution 2: override cache_hash method on resource to take associations in consideration
:::

```

File: docs/3.0/cache.md [hunk 3]
```diff
@@ -39,6 +39,14 @@ config.cache_store = ActiveSupport::Cache.lookup_store(:solid_cache_store)
 Our computed system do not use MemoryStore in production because it will not be shared between multiple processes (when using Puma).
 :::
 
+## Problem: cached rows on table do not update when record's associations suffer changes
```
Comment (adrianthedev) [inline docs/3.0/cache.md:42]: Let's not say "problem" in the title.

```suggestion
## Cached rows on the table do not update when the record's associations update
```

File: docs/3.0/cache.md [hunk 4]
```diff
@@ -39,6 +39,14 @@ config.cache_store = ActiveSupport::Cache.lookup_store(:solid_cache_store)
...
+:::warning
+
+Solution 1: touch: true on association
+
+Solution 2: override cache_hash method on resource to take associations in consideration
+:::
```
Comment (adrianthedev) [inline docs/3.0/cache.md:48]: We need to show examples of how that's achieved and not in a warning.

The structure is like so:

### Solution 1

use touch true..... <please fill in the rest>

```ruby
# example parent model with association

# example child model with association and touch true
```

### Solution 2

Avo, internally, uses the `cache_hash` method to compute the hash that will be remembered by the caching driver when displaying the rows.

You can take control and override it on that particular resource to take the association into account.

```ruby{5-13}
class Avo::Resources::User < Avo::BaseResource
  def fields
    # your fields
  end

  def cache_hash
    # record.post will now be taken under consideration 
    result = [record, file_hash, record.post]

    if parent_record.present?
      result << parent_record
    end

    result
  end
end
```

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Left some suggestions about how to change them to give our users more context and help.

**Comment by gabrielgiroe1** [comment]
We close this because it is related to #200

**Pull request closed without merging**